File: losowanie_liczb.pas
program losowanie_liczb;
{
 --== Borland Turbo Pascal 7.0 (Kurs podstawowy) ==--
 Copyright(c)by Jan T. Biernat
}
uses crt;
var A :Integer;
begin
  ClrScr;
  Writeln('--== Losowanie liczb ==--'); {Wyswietlenie na ekranie tekstu
                                         znajdujacego sie pomiedzy
                                         apostrofami.}
  Writeln('Copyright (c)by Jan T. Biernat');
  Writeln; {Przesuniecie kursora tekstowego o jeden wiersz w dol.}
  Randomize; {Zainicjowanie generatora liczb pseudolosowych.}
  A:= 0; {Wyzerowanie zmiennej liczbowej calkowitej "A".}
  A:= Random(99); {Wylosowanie liczby z podanego zakresu i przypisanie jej
                   do zmiennej liczbowej calkowitej "A".
                   Random(X) - Losuje liczbe z podanego zakresu liczb.
                               W miejsce parametru X nalezy umiescic
                               zmienna liczbowa calkowita lub
                               liczbe calkowita.
                               Wylosowana liczba bedzie z przedzialu od 0 do X-1.
                               Na przyklad:
                               Gdy w miejsce parametru X wpiszemy 11, to
                               maksymalna wylosowana liczba bedzie 10.}
  Writeln('Wylosowana liczba to: ', A); {Wyswietlenie na ekranie tekstu
                                         znajdujacego sie pomiedzy
                                         apostrofami - parametr 1
                                         oraz wyswietlenie zawartosci
                                         zmiennej liczbowej calkowitej "A" - parametr 2.
                                         Parametry oddzielane sa przecinkami.}
  Writeln;
  if(A mod 2 = 0) then {Jezeli reszta z dzielenia bedzie rowna zeru, to
                        wykonaj instrukcje po slowie THEN. W innym
                        przypadku wykonaj instrukcje po slowie ELSE.
                        MOD - zwraca reszte z dzielenia.}
     Writeln('Wylosowana liczba "', A, '" jest liczba parzysta.')
  else Writeln('Wylosowana liczba "', A, '" jest liczba nieparzysta.');
  {Zatrzymaj program do czasu nacisniecia klawisza ENTER.}
  Writeln; Writeln; Write('Nacisnij klawisz ENTER...');
  Readln;
end.

File: losowanie_liczb.cpp
/* 
 --== Borland C++Builder (Console Application) na C++ ==-- 
                               Kurs podstawowy 
 Copyright(c)by Jan T. Biernat 
*/ 
#include <iostream> 
#include <conio.h> 
#include <cstdlib> /*Posiada instrukcje do zainicjowania 
                     generatora liczb pseudolosowych.*/ 
#include <ctime> /*Biblioteka "ctime " zawiera funkcje 
                   związane z obsługą czasu i daty.*/ 
using namespace std; 
int main() { 
  cout << "--== Losowanie liczb ==--"; /*Wyświetlenie na ekranie 
                                         tekstu znajdującego 
                                         się pomiędzy cudzysłowami.*/ 
  const int Zakres = 34; /*Zadeklarowanie stałej liczbowej 
                           całkowitej "Zakres" i przypisanie 
                           jej konkretnej wartości (np. 34).*/ 
  srand(time(NULL)); //Zainicjowanie generatora liczb pseudolosowych. 
  int Liczba=0; /*Zadeklarowanie zmiennej liczbowej całkowitej 
                  "Liczba" z równoczesnym wyzerowaniem zmiennej.*/ 
  Liczba = rand()%Zakres; /*Wylosowanie jednej liczby z podanego zakresu, 
                            który przechowywany jest w stałej liczbowej 
                            całkowitej "Zakres" i przypisanie 
                            wylosowanej liczby do zmiennej liczbowej 
                            całkowitej "Liczba". 
                            Funkcja RAND - umożliwia wylosowanie jednej 
                                           liczby z podanego zakresu.*/ 
  cout << "\n\nWylosowano liczbe " << Liczba; /*Obniżenie kursora 
                                                tekstowego o dwa 
                                                wiersze i wyświetlenie 
                                                na ekranie tekstu 
                                                znajdującego się 
                                                pomiędzy cudzysłowami 
                                                oraz wyświetlenie 
                                                na ekranie zawartości 
                                                zmiennej liczbowej 
                                                całkowitej "Liczba". 
                                                Znak "\n" umożliwia 
                                                przeniesienie tekstu 
                                                do kolejnego wiersza. 
                                                Parametry oddzielane 
                                                są znakiem "<<".*/ 
  cout << "\nWylosowana liczba jest liczba "; 
  if(Liczba % 2 == 0) { cout << "parzysta."; } 
  else { cout << "nieparzysta."; } 
  /*"if(Liczba % 2 == 0)" - Jeżeli warunek jest spełniony 
                            (tj. zawartość zmiennej liczbowej 
                            całkowitej "Liczba" podzielonej 
                            przez dwa posiada po przecinku zero), 
                            to wyświetl informację "parzysta". 
                            W innym przypadku wyświetl 
                            informację "nieparzysta". 
                            Znak "%" - Umożliwia sprawdzenie liczby 
                                       po przecinku.*/ 
  //Naciśnij dowolny klawisz. 
  cout << "\n\nNacisnij dowolny klawisz..."; 
  getch(); 
} 
